fenix/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -216,17 +219,33 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
             } else {
                 0L
             }
-            AbstractFetchDownloadService.openFile(
+            val downloadState = DownloadState(
+                id = item.id,
+                url = item.url,
+                fileName = item.fileName,
+                contentType = item.contentType,
+                status = item.status,
+                contentLength = contentLength,
+            )
+
+            val onCannotOpenFile = AbstractFetchDownloadService.openFile(
                 applicationContext = it.applicationContext,
-                download = DownloadState(
-                    id = item.id,
-                    url = item.url,
-                    fileName = item.fileName,
-                    contentType = item.contentType,
-                    status = item.status,
-                    contentLength = contentLength,
-                ),
+                download = downloadState,
             )
+
+            if (!onCannotOpenFile) {
```

Comment:
the negation on onCannotOpenFile makes this a bit confusing (reads that it _can_ open) hence the name suggestion above

Reply:
Done